packages/android_alarm_manager/test/android_alarm_manager_test.dart

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:android_alarm_manager/android_alarm_manager.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  String invalidCallback(String foo) => foo;
+  void validCallback(int id) => null;
+
+  const MethodChannel testChannel = MethodChannel(
+      'plugins.flutter.io/android_alarm_manager', JSONMethodCodec());
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUpAll(() {
+    testChannel.setMockMethodCallHandler((MethodCall call) => null);
+  });
+
+  test('${AndroidAlarmManager.initialize}', () async {
+    testChannel.setMockMethodCallHandler((MethodCall call) async {
+      assert(call.method == 'AlarmService.start');
+      return true;
+    });
+
+    final bool initialized = await AndroidAlarmManager.initialize();
+
+    expect(initialized, isTrue);
+  });
+
+  group('${AndroidAlarmManager.oneShotAt}', () {
+    test('validates input', () async {
+      final DateTime validTime = DateTime.utc(1993);
+      final int validId = 1;
+
+      // Callback should take a single int param.
+      expect(
```

Comment:
You might want to update these test cases to use `expectLater` since these methods you're testing are `async`. This works at this point since we're hitting the asserts before any asynchronous code is run (async methods run synchronously until their first awaited operation), but if anything is added that performs an `await` things might not behave as you'd expect.

I typically use the following pattern for testing asynchronous code:

```dart
await expectLater(() async => await BadAssertAsynchronousCode(), throwsAssertionError);
```

Reply:
Good catch, fixed here and throughout.